### PR TITLE
snapstate: make sure all *link-*snap tasks carry a snap type and further hints

### PR DESCRIPTION
**Pull request by pedronis**
This makes sure all *link-*snap tasks carry the snap type, and also a hint whether the snap relevant revisions have only plugs.

This should help take decisions about conflicts and/or which operations can run together or not.

### CODE REVIEW

File: overlord/snapstate/snapmgr.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ type SnapSetup struct {
 	UserID  int       `json:"user-id,omitempty"`
 	Base    string    `json:"base,omitempty"`
 	Type    snap.Type `json:"type,omitempty"`
+	// PlugsOnly indicates whether the relevant revisions for the operation
+	// have only plugs or none, and absolutely no slots.
+	PlugsOnly bool `json:"plugs-only,omitempty"`
```
Comment (mvo5) [inline overlord/snapstate/snapmgr.go]: Should we use something like "Flags int" here or "Flags []string" ? But I guess its fine.

Comment (pedronis) [reply overlord/snapstate/snapmgr.go]: well we use structs of bools for other flags, don't think it make sense to add more nesting here

Comment (chipaca) [reply overlord/snapstate/snapmgr.go]: I'm not sure I parse the comment. Does it mean it has 0 slots and 0 or more plugs?

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -95,8 +95,15 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		return nil, err
 	}
 
-	// ensure core gets installed. if it is already installed return
-	// an empty task set
+	if snapst.IsInstalled() {
```
Comment (chipaca) [inline overlord/snapstate/snapstate.go]: why not `if snapsup.PlugsOnly && snapst.IsInstalled()` (and then just `snapsup.PlugsOnly = len(info.Slots) == 0` ?

File: overlord/snapstate/snapmgr.go (hunk 1 above)
Comment (pedronis) [reply overlord/snapstate/snapmgr.go]: yes

Comment (pedronis) [reply overlord/snapstate/snapmgr.go]: could have been called "no-slots",   but my brain seem to think better of "plugs-only" snaps than "no-slots" snaps,  also the former somehow sounds better

**Pull request merged**